### ISSUE TITLE
Entry editor: Text in entry row is not being wrapped

### ISSUE THREAD

**Issue by AEgit** [JabRef/jabref#5461]
JabRef 5.0.0-dev--2019-10-17----e81d58c51
Windows 10 10.0 amd64 
Java 12.0.2

The current entry editor does not wrap text in entry rows. This leads to a massive waste of screen space. Text in entry rows should be wrapped (= do an automatic return) to reduce the waste of screen space.

This problem has also been reported here:
https://github.com/JabRef/jabref/issues/5244#issuecomment-528031151

For examples of the waste of screen space see here:
https://github.com/JabRef/jabref/issues/5244#issuecomment-525290454
https://github.com/JabRef/jabref/issues/5244#issuecomment-528028140

**Comment by AEgit** [comment]
The following tickets are connected to this issue:
https://github.com/JabRef/jabref/issues/4479
https://github.com/JabRef/jabref/issues/555

The following changes might be have caused the problem in the first place:
https://github.com/JabRef/jabref/issues/4126 --> Title/author/instituition/year field were changed to single-line fields: In my opinion, at least the title and the author field should be multiline fields (but to be honest, I am not sure, whether these changes refer to the entry editor or to something else)

**Comment by matthiasgeiger** [comment]
The problem is more or less the difference between display and serialization: "Multiline fields" are fields that have a lot of contents and line breaks should be preserved during saving and loading (e.g. abstract). Other fields like author are per definition "single line fields" as there are no line breaks allowed. 

What it makes a bit difficult is the technical handling in the user interface when a "single line field" has to much content - as automated wrapping (at least in the past) also allowed for entering line breaks. However this should not be a big problem to be solved...

**Comment by rousselm** [comment]
I just installed 
JabRef 5.0--2020-03-06--2e6f433
Mac OS X 10.13.6 x86_64 
Java 13.0.2

I have for many years been using the standard LaTeX field annote to keep notes on my reading. I set up JabRef so that the annote field shows up in the Review tab instead of whatever the default is. (It's been so long since I set this up that I'm not sure anymore.) So "Set up general fields" says (among other things)
Review:annote

The problem is that annote doesn't wrap as I'm typing anymore. Instead, my notes end up on one long line (in the entry window) that scrolls off the edge of the screen. I suspect it's the same problem as above (or at least, in the same family), which is why I'm commenting here.

**Comment by ilippert** [comment]
> The problem is that annote doesn't wrap as I'm typing anymore. Instead, my notes end up on one long line (in the entry window) that scrolls off the edge of the screen. I suspect it's the same problem as above (or at least, in the same family), which is why I'm commenting here.

See https://github.com/JabRef/jabref/issues/4373 for bug of viewing and storing multiline fields

**Comment by tgkolda** [comment]
I am really having trouble viewing and editing author and title fields that go outside the box because they no longer wrap in the edit box. Is there a fix planned for this?

**Comment by AEgit** [comment]
The issue persists in:

JabRef 5.2--2020-12-09--d1fb9e2
Windows 10 10.0 amd64 
Java 14.0.2

**Comment by AEgit** [comment]
The issue persists in:

JabRef 5.3--2021-06-08--9d1c448
Windows 10 10.0 amd64 
Java 16.0.1 
JavaFX 16+8

**Comment by tgkolda** [comment]
Yes, the issue is persisting for me too, and I would very much like to see it resolved. For instance, the Title field does not wrap, but the abstract field does even though the "source" does not have line breaks. I'm using the latest official release...

JabRef 5.2--2020-12-24--6a2a512
Windows 10 10.0 amd64 
Java 14.0.2

**Comment by AEgit** [comment]
JabRef 5.6--2022-04-08--a6489cb
Windows 10 10.0 amd64 
Java 17.0.2 
JavaFX 18+12

The issue persists in the current dev version. Maybe as a workaround people can use the "BibTex source" tab and edit the author, title, etc. fields there directly. I reckon this is not really the way JabRef is supposed to be used (?), but otherwise it is impossible to edit entries that contain slightly longer titles or several authors.

**Comment by wujastyk** [comment]
JabRef 5.10--2023-02-01--611baac
Linux 5.15.0-58-generic amd64 
Java 19.0.2 
JavaFX 19+11

The "multiline" selector in "customize entry types" doesn't seem to do anything.  And if you return to that menu, the ticked box is unticked again.

**Comment by Siedlerchr** [comment]
@wujastyk  For me it works fine:

<img width="690" alt="grafik" src="https://user-images.githubusercontent.com/320228/217356442-8d8093e6-ac2f-458a-8a79-5c29eeced46a.png">

<img width="628" alt="grafik" src="https://user-images.githubusercontent.com/320228/217356601-9fe62c26-ed81-49bd-bdd5-6be01a8219e9.png">

**Comment by wujastyk** [comment]
@Siedlerchr Hmm.  It works for me too, in @book entries.  But I have a special database (used with Glossaries) that has only two fields.  And that doesn't work.  The setting tick-box just silently unticks itself and the field is never wrapped.  See attached video.

https://user-images.githubusercontent.com/762246/217361905-1f8865be-b40d-4937-889a-00c47ac21c93.mp4

**Comment by Siedlerchr** [comment]
Ah okay, thanks, that narrows it down. So it seems like it works for Standard Entry types, but not for custom entry types. 
I will try to investigate this in the next days!

**Comment by Siedlerchr** [comment]
@wujastyk  Found the issue and it's now solved (just merged)

**Comment by wujastyk** [comment]
Thank you! @Siedlerchr

**Comment by AEgit** [comment]
JabRef 5.10--2023-02-22--76b221d
Windows 10 10.0 amd64 
Java 19.0.2 
JavaFX 19+11

Note, that the original issue reported here still persists in the current JabRef dev version. The current workaround for the issue appears to be to set affected fields to "Multiline" in the "Customize entry types" menu. But I guess that is not the final solution (?).
